File: backend/trivia-data/src/main/java/com/delacruz/trivia/model/GameState.java
package com.ectrvia.trivia.model;

import lombok.Data;
import java.util.List;

@Data
public class GameState {
    private String roomCode;
    private String status;
    private Integer currentQuestionIndex;
    private Integer totalQuestions;
    private Question currentQuestion;
    private Integer timerSeconds;
    private Long questionStartTime;
    private List<LeaderboardEntry> leaderboard;
}

File: ECTriviaApp/app/src/main/java/com/delacruz/ectrivia/data/model/GameState.kt
package com.ectrvia.ectrivia.data.model

data class GameState(
    val eventType: GameEventType,
    val roomCode: String,
    val question: Question?,
    val questionIndex: Int,
    val totalQuestions: Int,
    val timerSeconds: Int,
    val questionStartTime: Long
)

enum class GameEventType {
    PLAYER_JOINED,
    PLAYER_LEFT,
    GAME_STARTING,
    QUESTION_START,
    QUESTION_END,
    GAME_FINISHED,
    HOST_CHANGED
}

data class AnswerResult(
    val isCorrect: Boolean,
    val correctAnswerIndex: Int,
    val pointsEarned: Int,
    val newTotalScore: Int,
    val newStreak: Int,
    val currentRank: Int
)

data class PlayerSession(
    val playerId: Long,
    val nickname: String,
    val roomCode: String,
    val isHost: Boolean
)
